internal/manifest/version.go

```diff
@@ -402,6 +404,14 @@ func (m *FileMetadata) ValidateVirtual(createdFrom *FileMetadata) {
 	if m.Size == 0 {
 		panic("pebble: virtual sstable size must be set upon creation")
 	}
+
+	if invariants.Enabled {
+		for key, _ := it.First(); key != nil; key, _ = it.Next() {
```

Comment:
The check we'd wanna do is that there's a point key at exactly the same user key as the lower bound, and another at exactly the upper bound, or a straddling range key/del at those bounds. That would confirm the bounds were generated as tight and there isn't a gap.

This check is really only mimicking the check that we're doing in the iterator stack.

Reply:
makes sense yeah i'll check that as well